src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [isOpen, setIsOpen] = useState(false);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [currentStatus, setCurrentStatus] = useState('all');
+  const [inputValue, setInputValue] = useState('');
+
+  const handleChangeCurrentStatus = (newCurrentStatus: string) =>
+    setCurrentStatus(newCurrentStatus);
+
+  const handleSetIsOpen = (value: boolean) => setIsOpen(value);
+
+  const handleSetCurrentTodo = (newTodo: Todo | null) =>
+    setCurrentTodo(newTodo);
+
+  const handleChangeInut = (newInputValue: string) =>
+    setInputValue(newInputValue);
+
+  const filteredTodos = (newStatus: string, qvery: string): Todo[] => {
+    return todos
+      .filter(todo => {
+        switch (newStatus) {
+          case 'all':
+            return true;
+
+          case 'active':
+            return !todo.completed;
+
+          case 'completed':
+            return todo.completed;
```

Comment:
Make Enum for 'all', 'active' and 'completed' instead of using hardcoded string. It's a more competent way of writing code and will also help to avoid errors